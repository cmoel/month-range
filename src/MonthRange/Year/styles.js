import glamorous from "glamorous";

const {div} = glamorous;

export const Months = div({
  alignContent: "center",
  alignItems: "center",
  display: "flex",
  flexFlow: "row wrap",
  padding: "inherit",
});

export const YearDiv = div({
  color: "#333",
  fontSize: "0.75em",
  fontWeight: 400,
  marginTop: "2em",
  paddingBottom: "0.5em",
});
