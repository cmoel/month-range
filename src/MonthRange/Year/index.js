import React, {Component} from "react";

import {Months, YearDiv} from "./styles";

class Year extends Component {
  render() {
    const {year, selectMonth, selection, children} = this.props;
    return (
      <article>
        <YearDiv>{year}</YearDiv>
        <Months>
          {React.Children.map(children, c =>
            React.cloneElement(c, {year, selection, selectMonth})
          )}
        </Months>
      </article>
    );
  }
}

export default Year;
