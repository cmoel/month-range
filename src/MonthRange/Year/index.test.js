import React from "react";
import Year from "./index";
import renderer from "react-test-renderer";

test("Year renders correctly", () => {
  const fst = undefined;
  const snd = undefined;
  const component = renderer.create(
    <Year year={2017} selectMonth={m => m} selection={{fst, snd}}>
      <div>Jan</div>
      <div>Feb</div>
      <div>Mar</div>
      <div>Apr</div>
      <div>May</div>
      <div>Jun</div>
    </Year>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
