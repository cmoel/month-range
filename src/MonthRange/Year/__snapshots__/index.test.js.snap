// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Year renders correctly 1`] = `
.glamor-0,
[data-glamor-0] {
  color: #333;
  font-size: 0.75em;
  font-weight: 400;
  margin-top: 2em;
  padding-bottom: 0.5em;
}

.glamor-1,
[data-glamor-1] {
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-flow: row wrap;
  padding: inherit;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-flow: row wrap;
}

<article>
  <div
    className="glamor-0"
  >
    2017
  </div>
  <div
    className="glamor-1"
  >
    <div
      selectMonth={[Function]}
      selection={
        Object {
          "fst": undefined,
          "snd": undefined,
        }
      }
      year={2017}
    >
      Jan
    </div>
    <div
      selectMonth={[Function]}
      selection={
        Object {
          "fst": undefined,
          "snd": undefined,
        }
      }
      year={2017}
    >
      Feb
    </div>
    <div
      selectMonth={[Function]}
      selection={
        Object {
          "fst": undefined,
          "snd": undefined,
        }
      }
      year={2017}
    >
      Mar
    </div>
    <div
      selectMonth={[Function]}
      selection={
        Object {
          "fst": undefined,
          "snd": undefined,
        }
      }
      year={2017}
    >
      Apr
    </div>
    <div
      selectMonth={[Function]}
      selection={
        Object {
          "fst": undefined,
          "snd": undefined,
        }
      }
      year={2017}
    >
      May
    </div>
    <div
      selectMonth={[Function]}
      selection={
        Object {
          "fst": undefined,
          "snd": undefined,
        }
      }
      year={2017}
    >
      Jun
    </div>
  </div>
</article>
`;
